server/utils/formUtils.ts

```diff
@@ -100,12 +101,12 @@ export function convertArrayToRadioItems(array: Array<string>, checkedItem: stri
 }
 
 /**
- * Performs validation on the area of a postcode (IE the first three characters)
+ * Performs validation on the area of a postcode (IE the first three or four characters)
  * @param string string to be validated.
  * @returns true if the string is valid, false otherwise.
  */
 export function validPostcodeArea(potentialPostcode: string) {
-  return /[A-Z][0-9]{1,2}|[A-Z][A-HJ-Y][0-9]{1,2}|[A-Z][0-9][A-Z]|[A-Z][A-HJ-Y][0-9]?[A-Z]/.test(potentialPostcode)
+  return postcodeAreas.includes(potentialPostcode.toUpperCase())
```

Comment:
I was worried that searching through a nearly 3000 entry array would slow down the tests but it doesn't seem to make too much different.